sql/core/src/test/scala/org/apache/spark/sql/Spark18601Suite.scala

```diff
@@ -0,0 +1,331 @@
+/*

```

Comment:
Move this into catalyst, and place this in the `org.apache.spark.sql.catalyst.optimizer`. I would call it the `complexTypesSuite.scala`